docs/src/main/asciidoc/grpc-getting-started.adoc

```diff
@@ -0,0 +1,318 @@
+                <execution>
+                    <id>compile</id>
+                    <goals>
+                        <goal>compile</goal>
+                        <goal>compile-custom</goal>
+                    </goals>
+                </execution>
+                <execution>
+                    <id>test-compile</id>
+                    <goals>
+                        <goal>test-compile</goal>
+                        <goal>test-compile-custom</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+
+        <!-- ... -->
+    </plugins>
+</build>
+----
+
+The `protobuf-maven-plugin` that generates stub classes from your gRPC service definition (`proto` files).
+The class generation uses a tool named `protoc`, which is OS-specific.
+That's why we use the `os-maven-plugin` to target the executable compatible with the operating system.
+
+NOTE: This configuration instructs the `protobuf-maven-plugin` to generate the default gRPC classes and classes using Mutiny to fit with the Quarkus development experience.
+
+With this configuration, you can put your service and message definitions in the `src/main/proto` directory.
+Let's start with a simple _Hello_ service.
+Create the `src/main/proto/helloworld.proto` file with the following content:
+
+[source]
+----
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "io.quarkus.example";
+option java_outer_classname = "HelloWorldProto";
+
+package helloworld;
+
+// The greeting service definition.
+service Greeter {
+    // Sends a greeting
+    rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+    string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+    string message = 1;
+}
+----
+
+This `proto` file defines a simple service interface with a single method (`SayHello`), and the exchanged messages (`HelloRequest` containing the name and `HelloReply` containing the greeting message).
+
+Before coding, we need to generate the classes used to implement and consume gRPC services.
+In a terminal, run:
+
+[source, bash]
+----
+$ mvn compile
+----
+
+Once generated, you can look at the `target/generated-sources/protobuf` directory:
+
+[source, txt]
+----
+target/generated-sources/protobuf
+├── grpc-java
+│   └── io
+│       └── quarkus
+│           └── example
+│               └── GreeterGrpc.java
+└── java
+    └── io
+        └── quarkus
+            └── example
+                ├── HelloReply.java
+                ├── HelloReplyOrBuilder.java
+                ├── HelloRequest.java
+                ├── HelloRequestOrBuilder.java
+                ├── HelloWorldProto.java
+                └── MutinyGreeterGrpc.java
+----
+
+These are the classes we are going to use.
+
+IMPORTANT: Every time you update the `proto` files, you need to re-generate the classes (using `mvn compile`).
```

Comment:
How does it work with hot reload?

Reply:
It does not work yet, @michalszynkiewicz is looking into that. The issue is that your code depends on these classes, so it must be done before the compilation.